src/layouts/Catalogue/Catalogue.js

```diff
@@ -1,358 +1,149 @@
-import { Modal, Button, InputNumber, } from "antd";
-import 'antd/dist/antd.css';
+import "antd/dist/antd.css";
 import styled from "styled-components";
-
-const loading = (
-  <div className="suspense-page">
-    {" "}
-    Loading...{" "}
-  </div>
-);
-
-export default class Catalogue extends Component {
-  state = {
-    menuOpened: false,
-    products: [],
-    selectedProduct: {},
-    onLoading: false,
-    showModal: false,
-    showModalCart: false,
-    cart: [],
-    qty: 1,
-  };
-
-  handleMenuOpen(e) {
-    this.setState({
-      menuOpened: !this.state.menuOpened,
-    });
+import useProductReducer from "../../components/CatalogueReducer";
+import useCartReducer from "../../components/CartReducer";
+
+const Wrapper = styled.div`
+  height: 100%;
+  width: 100%;
+`;
+
+const Main = styled.main`
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+  min-height: calc(100vh - 70px);
+  margin-top: 70px;
+  padding-top: 20px;
+  width: 100%;
+  background: white;
+`;
+
+const Title = styled.h1`
+  text-transform: uppercase;
+  font-size: 50pt;
+  font-weight: 700;
+  text-align: center;
+`;
+
+const Products = styled.ul`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: row;
+  flex-wrap: wrap;
+  width: 75%;
+`;
+
+export default function Catalogue() {
+  const [products, setProducts] = useState([]);
+  const [showModalCart, setShowModalCart] = useState(false);
+
+  const { state, actions } = useProductReducer();
+  const { stateCart, actionCarts } = useCartReducer();
+
+  useEffect(() => {
+    ProductApi.get()
+      .then((res) => res.json())
+      .then((res) => {
+        console.log(`res`, res);
+        setProducts(res);
+      });
+  }, []);
+
+  function showModal(product) {
+    actions.changeValue(product);
   }
 
-  getDataProducts() {
-    return ProductApi.get().then(res => res.json())
-    .then(res => {
-      this.setState({
-        products : res,
-      })
-    });
+  function handlerEmptyCart() {
+    actionCarts.reset();
   }
 
-  componentDidMount() {
-    document.title = "Katalog Produk";
-    this.getDataProducts();
+  function handleCancel() {
+    actions.reset();
   }
 
-  showModal(product){
-    this.setState({showModal: true, selectedProduct: product})
+  function handleCancelCart() {
+    setShowModalCart(false);
   }
 
-  handleOk = () => {
-    this.setState({ loading: true });
-    setTimeout(() => {
-      this.setState({ loading: false, showModal: false, selectedProduct: {} , qty: 1});
-    }, 3000);
-  };
-
-  handleCancel = () => {
-    this.setState({ showModal: false, selectedProduct: {}, qty: 1 });
-  };
-
-  handleOkCart = () => {
-    this.setState({ showModalCart: false });
-  };
-
-  handleCancelCart = () => {
-    this.setState({ showModalCart: false });
-  };
-
-  addtoCartHandler = () => {
-    var carts = this.state.cart
-    var qty = this.state.qty
+  function addtoCartHandler() {
+    var carts = stateCart.cart;
```

Comment:
`var` is not used anymore. Always use `const`. Only in some cases you need `let`.